packages/core/src/dialects/abstract/query-interface-typescript.ts

```diff
@@ -156,4 +164,45 @@ export class AbstractQueryInterfaceTypeScript {
   ): Promise<void> {
     await this.sequelize.queryRaw(this.queryGenerator.getToggleForeignKeyChecksQuery(enable), options);
   }
+
+  /**
+   * Splits a query string into multiple queries and executes them sequentially.
+   *
+   * @param query
+   * @param options
+   */
+  async executeAsSubQueries(query: string, options: QueryRawOptions) {
+    const results = [];
+    const subQueries = query.split(';').filter(subQuery => subQuery.trim().length > 0);
```

Comment:
What if a table name i.e. has a ; in the name? This possibly opens us up for injections and other unwanted behaviour.
Is it maybe possible to send over an array of SQL queries to be executed?

Reply:
I get it's already in the sqlite code, but if we're migrating that to TypeScript we should see if we can get a better solution than this

Reply:
I've rewritten this and added support for returning sql as an array of strings rather than a single string separated by semicolons.
Let me know if you're happy with the implementation.